kernel/include/ksched.h

```diff
@@ -383,16 +383,34 @@ int z_sched_wait(struct k_spinlock *lock, k_spinlock_key_t key,
  */
 void z_sched_usage_stop(void);
 
-void z_sched_usage_start(struct k_thread *thread);
+/**
+ * @brief Start thread cycle usage accounting
+ *
+ * This routine is called to mark the beginning of thread cycle usage
+ * and accounting.
+ *
+ * @param thread Identifies thread for which to track cycles
+ * @param new_window False unless a new thread is being switched in
+ */
+void z_sched_usage_start(struct k_thread *thread, bool new_window);
 
-uint64_t z_sched_thread_usage(struct k_thread *thread);
+/**
+ * @brief Retrieves cpu cycle usage data for specified core
+ */
+void z_sched_cpu_usage(uint8_t core_id, struct k_thread_runtime_stats *stats);
+
+/**
+ * @brief Retrieves thread cycle usage data for specified thread
+ */
+void z_sched_thread_usage(struct k_thread *thread,
+			  struct k_thread_runtime_stats *stats);
 
 static inline void z_sched_usage_switch(struct k_thread *thread)
 {
 	ARG_UNUSED(thread);
 #ifdef CONFIG_SCHED_THREAD_USAGE
 	z_sched_usage_stop();
-	z_sched_usage_start(thread);
+	z_sched_usage_start(thread, 1);
```

Comment:
Style note: second argument is declared as a bool, so should be passed as "true" and not "1".  I think there's some static analysis tool or MISRA checker or something that's picky about that.

Reply:
Will fix.